mcstatus/status_response.py

```diff
@@ -0,0 +1,640 @@
+    """Class for storing Java version info."""
+
+    @classmethod
+    def build(cls, raw: Dict[str, Any]) -> Self:
+        """Build `JavaStatusVersion` from raw response dict.
+
+        :param raw: Raw response dict.
+        :raise ValueError: If the required keys (name, protocol) are not present.
+        :raise TypeError: If the required keys (name - str, protocol - int) are not of the specified type.
+        :return: `JavaStatusVersion` object.
+        """
+        _validate_data(raw, "version", [("name", str), ("protocol", int)])
+        return cls(name=raw["name"], protocol=raw["protocol"])
+
+
+@dataclass
+class BedrockStatusVersion(StatusVersion):
+    """Class for storing Bedrock version info.
+
+    :param brand: Like `MCPE` or another.
+    """
+
+    brand: str
+
+
+class JavaStatusResponse(NewJavaStatusResponse):
+    """Class for storing JavaServer data from a status response.
+
+    :param icon: Icon of the server. Can be unset. BASE64 encoded.
+    """
+
+    class Players(JavaStatusPlayers):
+        """Deprecated class for `players` field.
+
+        Use `JavaStatusPlayers` instead.
+        """
+
+        class Player(JavaStatusPlayer):
+            """Deprecated class for player in `list` field.
+
+            Use `JavaStatusPlayer` instead.
+            """
+
+            @property
+            @deprecated(replacement="uuid", date="2022-08")
+            def id(self):
+                return self.uuid
+
+            @overload
+            def __init__(self, raw: Dict[str, Any]) -> None:
+                ...
+
+            @overload
+            def __init__(self, name: str, uuid: int) -> None:
+                ...
+
+            def __init__(self, *args, **kwargs) -> None:
+                try:
+                    bound = Signature(
+                        parameters=[Parameter("raw", Parameter.POSITIONAL_OR_KEYWORD, annotation=Dict[str, Any])]
+                    ).bind(*args, **kwargs)
+
+                    deprecated(
+                        lambda: None,
+                        replacement="mcstatus.status_response.JavaStatusPlayer",
+                        date="2022-08",
+                        display_name="JavaStatusResponse.Players.Player",
+                    )()
+
+                    # build and copy instance from `built` method
+                    super().__init__(**self.build(bound.arguments["raw"]).__dict__)
+                except TypeError:
+                    super().__init__(*args, **kwargs)
+
+        @overload
+        def __init__(self, raw: Dict[str, Any]) -> None:
+            ...
+
+        @overload
+        def __init__(self, online: int, max: int, list: List[JavaStatusPlayer]) -> None:
+            ...
+
+        def __init__(self, *args, **kwargs) -> None:
+            try:
+                bound = Signature(
+                    parameters=[Parameter("raw", Parameter.POSITIONAL_OR_KEYWORD, annotation=Dict[str, Any])]
+                ).bind(*args, **kwargs)
```

Comment:
Make all signature creation like statements these class variables. They don't need to be created each time `__init__` is called, since they don't need to be different for every instance. Creation on `__init__` only slows down the initialization.

Reply:
Fixed in 0b54716ff65436926e5b7889b9e165d42302170f.

Reply:
As I said, signatures should be moved into class variables, not as module globals:
```py
ABC = "Foo"  # Module global variable

class Foo:
   ABC = "Foo"  # Class variable
```
This signature will only be relevant for the given class, not for the entire file, there is no reason to define it as a file-wide global.

Reply:
This will increase number of initializations, because `JavaStatusResponse`, `JavaStatusResponse.Players` and `JavaStatusResponse.Players.Player` have the same old signature, so for all of the we need duplicate this signature.

Reply:
It's not that huge of an issue to create the same signature twice, or 3 times, it's just an issue to need to create it n times with each initialization of this class, of which there may be thousands of, since we don't know what the user may need to do with it.

That said, I didn't realize we would need to re-use some signatures somewhere, I would generally prefer to keep things localized though, and it probably does make more sense to make these local to the classes in this case, since they're only ever going to be relevant to those deprecated classes.

Also, no matter if they'll end up being global or class variables, they should be prefixed with `_` signifying that tit's only something we'll be using internally.

Reply:
This can be found with `dir()`, so I doesn't really want to move it. It is also can be prefixed with double `_`, so it will be more hard to use, but again, it can be found with `dir()`.

I really should prefix it with `_` (maybe even double underscore), and make it upper case.

> `SCREAMING_SNAKE_CASE`

This called as upper case, but okay.

Reply:
Renamed, but not moved, in 0139fa5b84e65fc737525a8144e96344e63d548d.